src/controllers/user.js

```diff
@@ -83,4 +83,60 @@ const logout = (req, res) => {
   }
 };
 
-module.exports = { login, register, logout };
+const createUser = async (req, res) => {
+  try {
+    const user = await User.create(req.body);
+    return res.status(201).json(user);
+  } catch (e) {
+    return res.status(400).json(e);
+  }
+};
+
+const getUser = async (req, res) => {
+  try {
+    const { id } = req.params;
+    const user = await User.findById(id);
+    if (!user) {
+      return res.status(404).json({ message: '404 Not found' });
+    }
+    return res.status(200).json(user);
+  } catch (e) {
+    return res.status(400).json(e);
+  }
+};
+
+const updateUser = async (req, res) => {
+  try {
+    const { id } = req.params;
+    const user = await User.findByIdAndUpdate(id, req.body, {
```

Comment:
again here sending the whole req.body is risky we need a $set maybe and destruct

Reply:
what if the user only needs to update his name?
 also sending req.body will only take the required parameters 
 it will not take the whole body
it took this approach from the assignments solutions
 and $set only if you want to update a nested document

Reply:
okay if you are sure that this will work why not keep it